hazelcast/src/test/java/com/hazelcast/client/console/ClientConsoleAppConfigTest.java

```diff
@@ -0,0 +1,90 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.console;
+
+import static com.hazelcast.test.AbstractHazelcastClassRunner.getTestMethodName;
+import static com.hazelcast.test.Accessors.getAddress;
+import static com.hazelcast.test.HazelcastTestSupport.assertTrueEventually;
+import static com.hazelcast.test.HazelcastTestSupport.smallInstanceConfig;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+
+import com.hazelcast.cluster.Address;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.OverridePropertyRule;
+import com.hazelcast.test.TestAwareInstanceFactory;
+import com.hazelcast.test.annotation.QuickTest;
+
+/**
+ * Tests a possibility to use a custom YAML client configuration file with the demo client console application.
+ */
+@RunWith(HazelcastSerialClassRunner.class)
+@Category({ QuickTest.class })
+public class ClientConsoleAppConfigTest {
+
+    @Rule
+    public TemporaryFolder tempFolder = new TemporaryFolder();
+
+    @Rule
+    public OverridePropertyRule propertyClientConfig = OverridePropertyRule.clear("hazelcast.client.config");
+
+    private final TestAwareInstanceFactory factory = new TestAwareInstanceFactory();
+
+    @After
+    public void after() {
+        factory.terminateAll();
+    }
+
+    @Test
+    public void connectsWhenYamlConfigSpecified() throws IOException {
+        HazelcastInstance hz = factory.newHazelcastInstance(smallInstanceConfig());
+        Address address = getAddress(hz);
+        File cfgFile = tempFolder.newFile("hazelcast-config.yml");
+        try (BufferedWriter writer = Files.newBufferedWriter(cfgFile.toPath())) {
+            writer.write("hazelcast-client:\n"
+                    + "  cluster-name: " + getTestMethodName() + "\n"
+                    + "  network:\n"
+                    + "    cluster-members:\n"
+                    + "      - " + address.getHost() + ":" + address.getPort() + "\n");
+        }
+        propertyClientConfig.setOrClearProperty(cfgFile.getAbsolutePath());
+        assertTrue(hz.getClientService().getConnectedClients().isEmpty());
```

Comment:
Consider removing this assertion. In general it's not a good practice to have assertions in the middle of the test, because you distract attention from what you'd like to test. And break the usual test pattern: `// given // when // then`. 

The same to `ConsoleAppConfigTest.connectsWhenYamlConfigSpecified()`.

Reply:
The assertion here is to avoid potential false positives. You can think about it as a part of the "given" clause. So I would rather keep it here.